presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -630,6 +675,34 @@ protected boolean internalStartNext()
         }
     }
 
+    protected void enforceTimeLimits()
+    {
+        checkState(Thread.holdsLock(root), "Must hold lock to enforce time limits");
+        synchronized (root) {
+            for (InternalResourceGroup group : subGroups.values()) {
+                group.enforceTimeLimits();
+            }
+            long runningTimeoutMillis = runningTimeout.toMillis();
+            long queuedTimeoutMillis = queuedTimeout.toMillis();
+            // Copy of activeQueries since queryFinished gets called by cancelQuery which modifies activeQueries
+            for (QueryExecution query : ImmutableList.copyOf(activeQueries)) {
+                if (runningQueries.contains(query)) {
+                    if (Optional.ofNullable(query.getQueryInfo().getQueryStats().getExecutionTime())
```

Comment:
Pull this out into a variable, and have the first if AND with it - no need for nesting.